lib/src/realm_class.dart

```diff
@@ -52,19 +52,24 @@ class Realm {
 
   /// Opens a `Realm` using a [Configuration] object.
   Realm(Configuration config) : _config = config {
-    _scheduler = Scheduler(config, close);
+    if (_config.isInUse) {
+      throw RealmException("A Realm instance for this configuraiton object already exists.");
+    }
+    _config.isInUse = true;
```

Comment:
I suggest we move this as the last operation of the constructor. Also this will remove the need to unset if there is an error

Reply:
Done